src/transformState.js

```
@@ -4,8 +4,35 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.keys(action.extraData).map(function(key) {
+          state[key] = action.extraData[key];
+        });
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.map(function(key) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        });
+        break;
 
+      case 'clear':
+        Object.keys(state).map(function(key) {
+          delete state[key];
+        });
+        break;
+
+      default:
+        break;
```

you should throw the Error in default case, like 'Unknown action type'